### PR TITLE
Specify from which version the API headers apply

### PR DESCRIPTION
**Pull request by zabil**
This is to clarify https://github.com/gocd/gocd/issues/1800

### CODE REVIEW

File: source/includes/_010-introduction.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ All APIs SHOULD be accessed from `https://go-server-url:8154/go/api`. All data S
 
 ## Current version
 
-By default, all requests receive **Version 1** of the API. It is recommended to explicitly request this version via the `Accept` header.
+Since v15.2.0,  all requests receive **Version 1** of the API. It is recommended to explicitly request this version via the `Accept` header.
```
Comment (ketan) [inline source/includes/_010-introduction.md]: Not all APIs have this header — Users may not be familiar with what APIs were introduced and when. I'd think rephrasing it to something along the lines of "Please check if the API request requires an `Accept` header with the api version" is probably better. WDYT?

**Comment by ketan** [comment]
Also — You'd need to submit this patch against 15.3 branch and 15.2 branch in addition to master.

**Comment by virtuald** [comment]
Dumb question: why not have all apis work with v1, regardless when they were specified (but of course, allow the 'older' ones to work without a v1 header)? Seems easier to explain, and less work for those trying to write API clients (though I suppose at this point it's already a mess).

File: source/includes/_010-introduction.md (hunk 1 above)
Comment (zabil) [inline source/includes/_010-introduction.md]: On second thought, 
Do the new API's work without the headers? If not, we should drop this section and just let the API section talk about it.

Comment (ketan) [inline source/includes/_010-introduction.md]: No the new APIs explicitly require a header. We can make them work without one, but that makes it ambiguous when we have to serve 2 different API versions at the same URL.

**Comment by ketan** [comment]
> Dumb question: why not have all apis work with v1, regardless when they were specified (but of course, allow the 'older' ones to work without a v1 header)?

That sends a false signal to users that the API is versioned when it really is (was?) not — which makes it a problem when we actually modernize and version some of the old APIs.

> Seems easier to explain, and less work for those trying to write API clients (though I suppose at this point it's already a mess).

I agree it is a bit of a mess — so I like @zabil's [idea](https://github.com/gocd/api.go.cd/pull/66#discussion_r49689637) of dropping the section altogether — can you think of a better way to do it?

File: source/includes/_010-introduction.md (hunk 1 above)
Comment (zabil) [inline source/includes/_010-introduction.md]: Yeah that's what I meant. The documentation is confusing.
'recommended to explicitly'

**Comment by virtuald** [comment]
Given the messiness, at the moment I think I would annotate each API call with "Minimum API Version required:", with values of N/A, v1, v2.

Then at the top, say something to the effect of "some api calls require a version. some don't. if a version is required, the API call will note the minimum version required"

It's messy, but at least it's correct.

**Pull request closed without merging**